bindings/rust/generate/src/main.rs

```diff
@@ -74,34 +117,79 @@ const PRELUDE: &str = r#"
 use libc::{iovec, FILE};
 "#;
 
-fn gen_bindings(entry: &str, s2n_dir: &Path, functions: FunctionCallbacks) -> bindgen::Builder {
+fn base_builder() -> bindgen::Builder {
     bindgen::Builder::default()
         .use_core()
         .layout_tests(true)
         .detect_include_paths(true)
         .size_t_is_usize(true)
-        .rustfmt_bindings(true)
-        .header_contents("s2n-sys.h", entry)
         .enable_function_attribute_detection()
         .default_enum_style(bindgen::EnumVariation::ModuleConsts)
-        .rust_target(bindgen::RustTarget::Stable_1_40)
-        // only export s2n-related stuff
-        .blocklist_type("iovec")
-        .blocklist_type("FILE")
-        .blocklist_type("_IO_.*")
-        .blocklist_type("__.*")
-        .blocklist_type("fpos_t")
+        .rust_target(bindgen::RustTarget::Stable_1_47)
         // rust can't access thread-local variables
         // https://github.com/rust-lang/rust/issues/29594
         .blocklist_item("s2n_errno")
         .raw_line(COPYRIGHT)
         .raw_line(PRELUDE)
         .ctypes_prefix("::libc")
+}
+
+fn gen_bindings(entry: &str, s2n_dir: &Path, functions: FunctionCallbacks) -> bindgen::Builder {
+    base_builder()
+        .header_contents("s2n-sys.h", entry)
+        // only export s2n-related stuff
+        .blocklist_type("iovec")
+        .blocklist_type("FILE")
+        .blocklist_type("_IO_.*")
+        .blocklist_type("__.*")
+        .blocklist_type("fpos_t")
         .parse_callbacks(Box::new(functions))
         .clang_arg(format!("-I{}/api", s2n_dir.display()))
         .clang_arg(format!("-I{}", s2n_dir.display()))
 }
 
+fn write_feature_bindings(
+    header_path: impl AsRef<Path>,
+    feature_flag: &str,
+    s2n_tls_sys_dir: &Path,
+    output_path: impl AsRef<Path>,
+    functions: FunctionCallbacks,
+) {
+    let header_path_str = format!("{}", header_path.as_ref().display());
+    let lib_path = s2n_tls_sys_dir.join("lib");
+    base_builder()
+        .header(&header_path_str)
+        .parse_callbacks(Box::new(
+            functions.with_feature(Some(feature_flag.to_owned())),
+        ))
+        // manually include header contents
+        .clang_arg(format!("-I{}/api", lib_path.display()))
+        .clang_arg(format!("-I{}", lib_path.display()))
+        .allowlist_recursively(false)
+        .allowlist_file(&header_path_str)
+        // s2n_internal.h defines opaque handles to these structs, but we want
+        // them to be imported from the main api module
+        .blocklist_type("s2n_connection")
+        .blocklist_type("s2n_config")
+        .raw_line("use crate::api::*;\n")
+        .generate()
+        .unwrap()
+        .write_to_file(output_path)
+        .unwrap();
+}
+
+fn header_name(header_path: impl AsRef<Path>) -> String {
+    let header_name_regex = Regex::new(r".*/(\w+)\.h").unwrap();
```

Comment:
I don't think we need a regex for this. Are you just wanting the name of the file?

https://doc.rust-lang.org/std/path/struct.Path.html#method.file_stem